server/services/mailer.service.js

```diff
@@ -61,7 +61,7 @@ export const sendMail = (defaultMailContent, user, additionalOptions = {}) => {
   const options = {
     ...defaultMailContent,
     ...additionalOptions,
-    firstName: user.firstName || null,
+    firstName: user.firstName || user.vendor.companyName ? user.vendor.companyName : null,
```

Comment:
I am not sure this will work as expected since `firstName` is a required field so it will always be present and the `companyName` will never be used here:

Not 100% sure too, but this looks better:
```suggestion
    firstName: user.vendor && user.vendor.companyName ? user.vendor.companyName : user.firstName || null,
```